File: dainf-lab-server/src/main/java/br/edu/utfpr/dainf/model/Reservation.java
package br.edu.utfpr.dainf.model;

import br.edu.utfpr.dainf.shared.Identifiable;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.List;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "reservation")
public class Reservation implements Identifiable<Long> {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "description")
    private String description;

    @Column(name = "observation")
    private String observation;

    @Column(name = "reservation_date")
    private Instant reservationDate;

    @Column(name = "withdrawal_date")
    private Instant withdrawalDate;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @NotNull(message = "Deve ser escolhido ao menos 1 produto.")
    @OneToMany(mappedBy = "reservation",
            cascade = {CascadeType.ALL}, orphanRemoval = true)
    private List<ReservationItem> items;
}

File: dainf-lab-client/src/app/pages/reservation/reservation.ts
import { Identifiable } from '@/shared/crud/crud';
import { User } from '../user/user';
import { Item } from '../item/item'; 

export interface ReservationItem extends Identifiable {
  item: Item; 
  quantity: number;
  price?: number; 
}

export interface Reservation extends Identifiable {
  description?: string;
  observation?: string;
  reservationDate: Date;
  withdrawalDate?: Date;
  user: User;
  items: ReservationItem[];
}